### PR TITLE
Add winstonjs as a logging library

### PR DESCRIPTION
**Pull request by jcooky**
This PR is to apply winstonjs as a logger
It depends on https://github.com/tymsai/marmoym-config/pull/5

## References
- https://github.com/tymsai/marmoym-api/issues/70

### CODE REVIEW

File: src/controllers/Definition/DefinitionController.ts [hunk 1]
```diff
@@ -1,20 +1,22 @@
-import { db1 } from '../../database';
+import {db1} from '../../database';
```
Comment (enginehenryed) [inline src/controllers/Definition/DefinitionController.ts]: add spaces inside curly braces https://github.com/airbnb/javascript#whitespace--in-braces

File: src/controllers/Definition/DefinitionController.ts [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
  * ...
  */
 export async function getDefinitionByTermId(termId: any) {
   var list = await db1.definition.findAll({
-    where : {
-      status : {$not: "DELETED"},
-      $or : [
-        {term_id : termId}
+    where: {
+      status: {$not: "DELETED"},
```
Comment (enginehenryed) [inline src/controllers/Definition/DefinitionController.ts]: add spaces inside curly braces

File: src/controllers/Definition/DefinitionController.ts [hunk 3]
```diff
@@ -1,20 +1,22 @@
...
+      $or: [
+        {term_id: termId}
```
Comment (enginehenryed) [inline src/controllers/Definition/DefinitionController.ts]: add spaces inside curly braces

File: src/app.ts [hunk 4]
```diff
@@ -6,32 +6,36 @@ import * as express from 'express';
...
-// import morgan from 'morgan';
+import * as https from 'https';
+import * as winston from 'winston';
 
 import config from './config'
 import * as errorHandleService from './services/errorHandleService';
 
 import routes from './routes';
 import database from './database';
+import * as fs from "fs";
```
Comment (enginehenryed) [inline src/app.ts]: external dependencies (fs, path, cors ...) to be written in the upper

File: src/constants/ErrorType/UserError.ts [hunk 5]
```diff
@@ -6,5 +6,13 @@ export default {
   "USER_NOT_FOUND": {
     code: 401000,
     msg: "text" // TODO: to be changed soon
+  },
+  "INVALID_TOKEN": {
```
Comment (enginehenryed) [inline src/constants/ErrorType/UserError.ts]: `TOKEN_INVALID`, I'd suggest that the subject (noun) comes first followed by verb (or adjective)

File: src/constants/ErrorType/UserError.ts [hunk 6]
```diff
@@ -6,5 +6,13 @@ export default {
...
+    code: 401002,
+    msg: "Invalid token"
+  },
+  "NOT_EQUAL_USERNAME": {
```
Comment (enginehenryed) [inline src/constants/ErrorType/UserError.ts]: How about `USER_USERNAME_NOT_EQUAL` or `USERNAME_NOT_EQUAL`?

File: src/services/errorHandleService.ts [hunk 7]
```diff
@@ -1,15 +1,17 @@
+import * as winston from 'winston';
+
 export const handleError = (err, req, res, next) => {
   if (err.constructor.name == 'MarmoymError') {
-    console.log(err);
+    winston.warn(err);
```
Comment (enginehenryed) [inline src/services/errorHandleService.ts]: Just a question: why is it `warn` when others are `error`?

File: src/app.ts [hunk 8]
```diff
@@ -6,32 +6,36 @@ import * as express from 'express';
 import * as cors from 'cors';
 import * as bodyParser from 'body-parser';
 import * as path from 'path';
-// import morgan from 'morgan';
+import * as https from 'https';
+import * as winston from 'winston';
 
 import config from './config'
 import * as errorHandleService from './services/errorHandleService';
```
Comment (enginehenryed) [inline src/app.ts]: unnecessary blank line

**Comment by enginehenryed** [comment]
https://github.com/tymsai/marmoym-api/pull/71#issue-253059576
comment has been modified
`It needs acceptance of -` => `It depends on`

**Comment by enginehenryed** [comment]
In commit message, `Add wintonsjs as a log library` would be better. We don't put article in front of proper noun.

File: src/app.ts (hunk 4 above)
Comment (jcooky) [reply src/app.ts]: 왜 외부 의존 라이브러리에 대해서 Upper로 써야하는건가요?

File: src/services/errorHandleService.ts (hunk 7 above)
Comment (jcooky) [reply src/services/errorHandleService.ts]: 우리가 핸들링하고 있는 에러들은 warning으로 하였고, 핸들링 되지 않는 에러인 Internal Error의 경우 error로 하였습니다.

**Comment by jcooky** [comment]
@enginehenryed Thank you for your comments

File: src/controllers/Definition/DefinitionController.ts (hunk 1 above)
Comment (jcooky) [reply src/controllers/Definition/DefinitionController.ts]: 수정하겠습니다.

File: src/controllers/Definition/DefinitionController.ts (hunk 2 above)
Comment (jcooky) [reply src/controllers/Definition/DefinitionController.ts]: 수정하겠습니다.

File: src/controllers/User/UserSignInController.ts [hunk 9]
```diff
@@ -16,7 +16,7 @@ export const signInUser = async (userInfo) => {
       }
     })
     .catch(() => {  // TODO: 이 catch 문은 차후 에러 오브젝트를 디자인하고 수정해야함
-      throw new MarmoymError(ErrorType.USER_NOT_FOUND);
+      throw new MarmoymError(UserError.USER_NOT_FOUND);
```
Comment (enginehenryed) [inline src/controllers/User/UserSignInController.ts]: This was talked in offline meeting. Please try to use ErrorType for any type of Error. That is what ErrorType exist for. For every type of Error, as long as it is MarmoymError, we can use `ErrorType`.

File: src/services/authService.ts [hunk 10]
```diff
@@ -14,13 +13,13 @@ const _verifyUserToken = async (token: string, username: any) => {
     decoded = jwt.verify(token, config.auth.jwtSecret);
     winston.debug('JWT decoded: ', decoded);
   } catch(err) {
-    throw UserError.INVALID_TOKEN;
+    throw new MarmoymError(UserError.TOKEN_INVALID);
```
Comment (enginehenryed) [inline src/services/authService.ts]: Check out my comment elsewhere

File: src/routes/v1/testRoute.ts [hunk 11]
```diff
@@ -22,7 +22,7 @@ router.get('/', tokenAuthHandler, async (req: Request, res: Response) => {
 });
 
 router.get("/a", (req, res) => {
-  throw new MarmoymError(ErrorType.USER_INCORRECT_PASSWORD);
+  throw new MarmoymError(UserError.USER_INCORRECT_PASSWORD);
```
Comment (enginehenryed) [inline src/routes/v1/testRoute.ts]: Checkout my comment elsewhere

File: src/controllers/User/UserSignInController.ts (hunk 9 above)
Comment (jcooky) [reply src/controllers/User/UserSignInController.ts]: 이 리뷰와 나머지 코멘트에 대해서도 수정하겠습니다~

**Pull request merged**